torch/_refs/__init__.py

```diff
@@ -929,6 +933,37 @@ def sub(
     aten_op=None,  # CompositeImplicitAutograd
 )
 
+
+def _xlogy(a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]):
+    assert isinstance(a, TensorLike) or isinstance(b, TensorLike)
+
+    # 1) Operations derived from elementwise_unary_scalar_wrapper contain non-ref
+    # functions, which causes a runtime error in TorchRefsMode context manager
+    # when strict = True
+    # 2) where requires all tensors to be on the same device
+    # TODO remove scalar_tensor
+    if isinstance(a, TensorLike):
+        if isinstance(b, Number):
+            b = scalar_tensor(b, dtype=a.dtype, device=a.device)
```

Comment:
I don't think this will handle the type promotion properly:

```
torch.xlogy(torch.tensor((1, 2, 3)), 5.)
: tensor([1.6094, 3.2189, 4.8283])
```

Reply:
But type promotion will already be done in the wrapper.... hmm....

Reply:
Does type promotion come before or after a scalar becomes a tensor in aten?

Reply:
Before